outline/tun2socks/tunnel_android.go

```diff
@@ -52,3 +54,29 @@ func ConnectShadowsocksTunnel(fd int, client *shadowsocks.Client, isUDPEnabled b
 	go tunnel.ProcessInputPackets(t, tun)
 	return t, nil
 }
+
+// ConnectTunnel reads packets from a TUN device represented by `fd` and routes
+// them to a remote proxy server represented by `configJSON`. This function
+// will also do connectivity tests before starting, so the caller is not
+// required to do any UDP tests.
+//
+// If the function succeeds, it will return a nil error and a Tunnel instance.
+//
+// This function does *not* take ownership of the TUN file descriptor `fd`; the
+// caller is responsible for closing after Tunnel disconnects.
+func ConnectTunnel(configJSON string, fd int) (Tunnel, error) {
```

Comment:
The ConnectTunnel function is doing different things in different platforms. That is confusing. We need to fix that.

Can you make this take a ReadWriteCloser for the tun device and move this to the shared code?